flycheck-pos-tip.el

```
@@ -93,11 +96,15 @@ messages on TTY frames if `flycheck-pos-tip-mode' is active."
                         ;; pos-tip computes the offset from the top of the line
                         ;; apparently.
                         nil (and line-height (+ line-height 5))))
-      (funcall flycheck-pos-tip-display-errors-tty-function errors))))
+      (funcall flycheck-pos-tip-display-errors-tty-function errors))
+    (dolist (hook flycheck-pos-tip--hide-hooks)
+      (add-hook hook #'flycheck-pos-tip-hide-messages))))
 
 (defun flycheck-pos-tip-hide-messages ()
   "Hide messages currently being shown if any."
   (unless (flycheck-pos-tip--check-pos)
+    (dolist (hook flycheck-pos-tip--hide-hooks)
+      (remove-hook hook #'flycheck-pos-tip-hide-messages))
```

Shouldn't this be out of the `unless` block?

---

No, that doesn't quite work.  `flycheck-pos-tip-hide-messages` is called in post-command-hook so invoking any command that doesn't happen to move point would cause the hook to be removed if it was outside of the `unless` block.

Consider the following example operations:

1. Move point onto a highlighted error.  Wait a bit, then the tooltip will be displayed.
2. Press `C-h .` (or even just `C-g`, etc.) which is a command that does not change current-buffer, buffer-modified-tick, or point
3. Move point off of the error.  This should hide the tooltip

With the suggested snippet moved outside of the `unless` block, (3) does not hide the error as it should.

(Sorry for the delay, I had a lot of personal stuff going on recently.)